### ISSUE TITLE
pre-compiled Projucer binaries fail to run on Debian/amd64

### ISSUE THREAD

**Issue by umlaeute** [WeAreROLI/JUCE#423]
downloading [JUCE for Linux](https://d30pueezughrda.cloudfront.net/juce/juce-5.3.2-linux.zip) on my Debian/buster amd64 system, I cannot run the pre-compiled binary `Projucer`:

~~~sh
$ wget https://d30pueezughrda.cloudfront.net/juce/juce-5.3.2-linux.zip
$ unzip juce-5.3.2-linux.zip
$ JUCE/Projucer
JUCE/Projucer: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by JUCE/Projucer)
$
~~~

btw, `DemoRunner` starts nicely.

Self-compiling Projucer produces a working binary (but is impractical under certain circumstances, like continuous-integration where you don't want to spend time compiling your build-system)

`libcurl3` is being faded out of Debian/Ubuntu;
while (on older distros) there is a `libcurl3` package which can be used to satisfy the dependencies to run `Projucer`.
however, there are no more -dev packages (all have switched to `libcurl4` - **conflicting** with `libcurl3`), making it impossible to *compile* any JUCE-plugins at the same time as having a running `Projucer`.

I don't know the exact state on recent Ubuntu releases (being a Debian guy), but I expect problems there as well (so probably a lot of Linux devs are bitten by this)

**Comment by BennetLeff** [comment]
I've had the same behaviour with the download for Linux (running on Manjaro 17). I'm going to give it a shot compiling the library from the source on Github. Have you figured this out yet @umlaeute @miccio-dk ?

**Comment by BennetLeff** [comment]
I've found a solution! Thanks to Fabian in the JUCE forum. Here's a copy-paste of Fabian's answer:

Just re-compile the Projucer yourself.

First install some build-time dependencies (this is for debian or ubuntu - may be different on your distro)
and then compile the Projucer:

    cd JUCE/extras/Projucer/Builds/LinuxMakefile
    make -j4 CONFIG=Release
The Projucer will then be in `JUCE/extras/Projucer/Builds/LinuxMakefile/builds.`

As far as I can tell, this can now be closed.

**Comment by umlaeute** [comment]
@BennetLeff i don't agree that the issue can be closed.

the title is quite specific as the problem is with  "pre-compiled JUCE" (as opposed to "self-compiled").

(and yes, i'm quite familiar with self-compiling Projucer)

**Comment by miccio-dk** [comment]
Compiling the project from scratch solved my issue as well. However, as @umlaeute pointed out, the issue with pre-compiled distribution remains.

**Comment by LGoodacre** [comment]
There is also the point that it won't compile without libcurl3.

**Comment by umlaeute** [comment]
@LGoodacre are you sure?
The Debian packages are compiled with libcurl4 (`libcur4-gnutls-dev` to be precise), and there doesn't seem to be an issue with *that*.

**Comment by BennetLeff** [comment]
Regarding not closing the issue - that's a fair point, I was mostly just happy I found an easy fix and got carried away. Hopefully, other devs see this issue and realize they can just rebuild until an actual PR fixes it.

**Comment by LGoodacre** [comment]
@umlaeute Yes, I'm pretty sure. With libcurl4 installed, I was trying to compile Proucer without success, based off of the same forum post that @BennetLeff mentioned. Then I came across this thread and did `sudo apt install libcurl3` and finally got Projucer to compile.

**Comment by tpoole** [comment]
This should be fixed in JUCE 5.4.1 - we now load the libcurl functions we need at runtime.
